packages/astro/src/@types/astro.ts

```diff
@@ -3012,83 +3012,92 @@ export type HookParameters<
 	Fn = AstroIntegration['hooks'][Hook],
 > = Fn extends (...args: any) => any ? Parameters<Fn>[0] : never;
 
+declare global {
+	// eslint-disable-next-line  @typescript-eslint/no-namespace
+ 	namespace Astro {
```

Comment:
What is the advantage of a namespace over a module declaration?

Reply:
Two:
- Makes a distinction between types we are intentionally allowing third-party libraries to extend vs types that just happen to be interfaces at the root level and, as such, are extendable.
- We already use namespaces for this one the `Astro.locals`, though there it uses the `App` namespace instead of the `Astro` namespace.

Reply:
Would be nice to use the same namespace everywhere, I'd be curious to know why it was named App

Reply:
I suppose so it wouldn't be ambiguous on `.astro` files when using the compiler-injected constant vs the namespace

Reply:
I believe it makes sense to separate `Astro` and `App`. `App` for end-users, while `Astro` works across all Astro projects that use a particular integration. 

In fact. `Locals` change **per project**. However, `IntegrationHooks` change **per integration**, which means multiple projects will have the same type.